supabase/migrations/37_alerts.sql

```diff
@@ -0,0 +1,136 @@
+create table alert_subscriptions (
+  like internal._model including all,
+
+  catalog_prefix    catalog_prefix                 not null,
+  email             text
+);
+alter table alert_subscriptions enable row level security;
+
+create policy "Users access subscriptions for the prefixes they admin"
+  on alert_subscriptions as permissive
+  using (exists(
+    select 1 from auth_roles('admin') r where catalog_prefix ^@ r.role_prefix
+  ));
+
+grant select, insert, update, delete on alert_subscriptions to authenticated;
+
+create table alert_data_processing (
+  catalog_name           catalog_name not null,
+  evaluation_interval    interval     not null,
+  primary key (catalog_name)
+);
+alter table alert_data_processing enable row level security;
+
+create policy "Users access alerts for admin-authorized tasks"
+  on alert_data_processing as permissive
+  using (exists(
+    select 1 from auth_roles('admin') r where catalog_name ^@ r.role_prefix
+  ));
+
+grant update (evaluation_interval) on alert_data_processing to authenticated;
+grant select, insert, delete on alert_data_processing to authenticated;
+
+create table alert_history (
+  alert_type      text         not null,
+  catalog_name    catalog_name not null,
+  fired_at        timestamptz  not null,
+  resolved_at     timestamptz,
+  arguments       json         not null,
+  primary key (alert_type, catalog_name)
+);
+alter table alert_history enable row level security;
+
+create policy "Users access alert history for admin-authorized tasks"
+  on alert_history as permissive
+  using (exists(
+    select 1 from auth_roles('admin') r where catalog_name ^@ r.role_prefix
+  ));
+
+grant select on alert_history to authenticated;
+
+create view internal.alert_data_processing_firing as
+select
+  alert_data_processing.*,
+  'data_not_processed_in_interval' as alert_type,
+  alert_subscriptions.email,
+  live_specs.spec_type,
+  coalesce(sum(catalog_stats_hourly.bytes_written_by_me + catalog_stats_hourly.bytes_written_to_me + catalog_stats_hourly.bytes_read_by_me), 0)::bigint as bytes_processed
+from alert_data_processing
+  left join live_specs on alert_data_processing.catalog_name = live_specs.catalog_name and live_specs.spec is not null and (live_specs.spec->'shards'->>'disable')::boolean is not true
+  left join catalog_stats_hourly on alert_data_processing.catalog_name = catalog_stats_hourly.catalog_name
+  left join alert_subscriptions on alert_data_processing.catalog_name ^@ alert_subscriptions.catalog_prefix
+where live_specs.created_at <= date_trunc('hour', now() - alert_data_processing.evaluation_interval)
+      and catalog_stats_hourly.ts >= date_trunc('hour', now() - alert_data_processing.evaluation_interval)
```

Comment:
I think this `catalog_stats_hourly` condition needs to be a part of the `join` condition instead of the `where`. If this were an inner join, there would be no logical difference. But we need this to be an outer join so that a row is returned even for tasks that have no corresponding `catalog_stats_hourly` rows. Ideally, this could be covered by a test case where there's a `live_specs` row but no corresponding `catalog_stats_hourly` rows within the time window.